batchspawner/batchspawner.py

```diff
@@ -467,20 +480,20 @@ class BatchSpawnerRegexStates(BatchSpawnerBase):
         If this variable is set, the match object will be expanded using this string
         to obtain the notebook IP.
         See Python docs: re.match.expand""").tag(config=True)
+    state_unknown_re = Unicode('^$',
+        help="Regex that matches job_status if the resource manager is not answering").tag(config=True)
 
     def state_ispending(self):
         assert self.state_pending_re, "Misconfigured: define state_running_re"
-        if self.job_status and re.search(self.state_pending_re, self.job_status):
-            return True
-        else:
-            return False
+        return self.job_status and re.search(self.state_pending_re, self.job_status)
 
     def state_isrunning(self):
         assert self.state_running_re, "Misconfigured: define state_running_re"
-        if self.job_status and re.search(self.state_running_re, self.job_status):
-            return True
-        else:
-            return False
+        return self.job_status and re.search(self.state_running_re, self.job_status)
+
+    def state_isunknown(self):
+        assert self.state_unknown_re, "Misconfigured: define state_unknown_re"
```

Comment:
```suggestion
        if not self.state_unknown_re:
            return False
```
I guess this doesn't need to be implemented for all spawners... it goes back to default behavior if it's not here.